src/prefect/runner/storage.py

```diff
@@ -276,14 +279,204 @@ def to_pull_step(self) -> Dict:
+        local storage. If None, remote storage will perform a one-time sync.
+        """
+        return self._pull_interval
+
+    @property
+    def destination(self) -> Path:
+        """
+        The local file path to pull contents from remote storage to.
+        """
+        return self._storage_base_path / self._remote_path
+
+    @property
+    def _remote_path(self) -> Path:
+        """
+        The remote file path to pull contents from remote storage to.
+        """
+        _, netloc, urlpath, _, _ = urlsplit(self._url)
+        return Path(netloc) / Path(urlpath.lstrip("/"))
+
+    async def pull_code(self):
+        """
+        Pulls contents from remote storage to the local filesystem.
+        """
+        self._logger.debug(
+            "Pulling contents from remote storage '%s' to '%s'...",
+            self._url,
+            self.destination,
+        )
+
+        if not self.destination.exists():
+            self.destination.mkdir(parents=True, exist_ok=True)
+
+        remote_path = str(self._remote_path) + "/"
+
+        try:
+            await from_async.wait_for_call_in_new_thread(
+                create_call(
+                    self._filesystem.get,
+                    remote_path,
+                    str(self.destination),
+                    recursive=True,
+                )
+            )
+        except Exception as exc:
+            raise RuntimeError(
+                f"Failed to pull contents from remote storage {self._url!r} to"
+                f" {self.destination!r}"
+            ) from exc
+
+    def to_pull_step(self) -> dict:
+        """
+        Returns a dictionary representation of the storage object that can be
+        used as a deployment pull step.
+        """
+
+        def replace_block_with_placeholder(obj: Any) -> Any:
+            if isinstance(obj, Block):
+                return f"{{{{ {obj.get_block_placeholder()} }}}}"
+            return obj
+
+        settings_with_placeholders = visit_collection(
+            self._settings, replace_block_with_placeholder, return_data=True
+        )
+        requires = self._get_required_package_for_scheme(urlparse(self._url).scheme)
+        step = {
+            "prefect.deployments.steps.pull_from_remote_storage": {
+                "url": self._url,
+                **settings_with_placeholders,
+            }
+        }
+        if requires:
+            step["prefect.deployments.steps.pull_from_remote_storage"][
+                "requires"
+            ] = requires
+        return step
+
+    def __eq__(self, __value) -> bool:
+        """
+        Equality check for runner storage objects.
+        """
+        if isinstance(__value, RemoteStorage):
+            return self._url == __value._url and self._settings == __value._settings
+        return False
+
+    def __repr__(self) -> str:
+        return f"RemoteStorage(url={self._url!r})"
+
+
 def create_storage_from_url(
```

Comment:
do you think there'd be any value in passing kwargs through here so that you could get settings from `from_source` to `RunnerStorage`?